cmd/honeycloudtrail/main.go

```diff
@@ -0,0 +1,232 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"reflect"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudtrail"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/honeycombio/honeyaws/logbucket"
+	"github.com/honeycombio/honeyaws/options"
+	"github.com/honeycombio/honeyaws/publisher"
+	"github.com/honeycombio/honeyaws/state"
+	libhoney "github.com/honeycombio/libhoney-go"
+	flag "github.com/jessevdk/go-flags"
+)
+
+var (
+	opt        = &options.Options{}
+	BuildID    string
+	versionStr string
+)
+
+// checks to see if struct contains a field
+// Trail type will not contain certain fields we want to check
+// such as S3KeyPrefix and S3BucketName if they are not set
+// causing a panic
+func nilField(v interface{}, name string) bool {
```

Comment:
Can you explain a little bit more why not just check for nil like `trail.S3KeyPrefix == nil` ?